sentrework.tex

```diff
@@ -0,0 +1,230 @@
+weak output iterators are consistent with those single-pass semantics. The
+distinction is obviously support for equality comparison. How do algorithms use
+equality comparison with iterators for single-pass ranges?
+
+\pnum
+Oddly enough, the answer to the question posed is that algorithms need not ever
+use equality comparison on single-pass iterators. In the iterator pair model its
+necessary to compare an iterator denoting the current position in the range with
+the supplied end iterator to determine when the algorithm reaches the end of the
+range. An algorithm \textit{could} compare the ``current'' iterator with itself
+or a copy thereof, but it would be pointless to do so. An algorithm
+\textit{cannot} compare the ``current'' iterator with a copy of an earlier value
+it held because that operation is not required to be valid for single-pass
+iterators.
+
+\pnum
+The case is similar for the iterator / sentinel model. An algorithm compares the
+iterator with the sentinel to determine when the end of the range is reached,
+but cannot compare the iterator with a copy of an older value. It would again be
+pointless to compare the iterator with itself or a copy; there's no reason to
+test things known to be equal for equality. Consequently, relaxing the
+algorithms that use single-pass ranges to ``weak'' single-pass ranges doesn't
+reduce their power.
+
+\rSec2[sentrework.ram.merge]{Why ``strong'' Iterators?}
+
+\pnum
+Relaxing the algorithms necessitates relaxing the iterator operations
+(\tcode{advance} et al~(\ref{iterator.operations})) which the algorithms use.
+Indeed there are many components in the specification that require a ``strong''
+iterator not because they need to compare iterator values for equality, but
+because iterator / sentinel pairs in N4560 necessarily used strong iterators.
+Once all such occurrences are relaxed it becomes apparent that there is no need
+for separate ``weak'' and ``strong'' concepts for input and output iterators,
+the strong concepts are unused. The current design removes N4560's
+\tcode{InputIterator} and \tcode{OutputIterator} concepts and strips the
+\tcode{Weak} prefix from the names of N4560's \tcode{WeakInputIterator} and
+\tcode{WeakOutputIterator} since the name distinction is no longer necessary.
+
+\pnum
+After removing the distinction between weak and strong input and output
+iterators, the \tcode{Iterator} concept is insufficiently distinguished from
+\tcode{Sentinel} to deserve a separate named concept. The current design pushes
+\tcode{Iterator}'s \tcode{Sentinel<I,I>()} requirement up into
+\tcode{ForwardIterator} so that the semantics of forward and stronger iterators
+are unchanged, and pushes the \tcode{EqualityComparable} requirement from
+\tcode{Iterator} down into \tcode{Sentinel}, but only requires it when the
+iterator and sentinel types are the same. \tcode{Iterator} may then be struck
+from the design, and the \tcode{Weak} prefix stripped from \tcode{WeakIterator}
+as well.
```

Comment:
`WeakIterator` is now `Iterator`, which aligns nicely with the usage of "iterators" in the prose.